### DISCUSSION TITLE
does computer vision need some help?

### DISCUSSION THREAD

**Discussion by eaallen** [hiukim/mind-ar-js#160] [General]
Hello @hiukim. I am student at Brigham Young University. I really like mindAR and I would love to see it improve. I have purchased your udemy course I am loving all the content. 

In the course, you mentioned that the **computer vision** for mindAR is under par. I am wondering if this is something you would like to have improved?  I do not know anything about computer vision, but I may have connections who do and would be interested in contributing to this project. 

Anyway, I am wounding what the known issues and pain points are for mindAR's computer vision. 

Hoping to help in anyway I can,

-eaallen

**Comment by hiukim** [comment]
Hi @eaallen, yes absolutely. Please let me know if anyone is interested in helping with computer vision! That would be super useful.

Top of my head, there are couple of things:
I feel like some of my algorithms implementation didn't done right. I have some knowledge gap on the mathematics. Even having someone to review certain part of the code would be nice.

I feel like the tracking part can be done better, maybe with better approaches or algorithms. Now, it appears jittery. and the frame rate could be better. I would say, the detection part seems relatively solid, but frame rate could also improve. 

adding new features. World tracking is definitely something interesting to a lot of people. So if someone is able to bring SLAM into web, we could potentially make that happens.
